src/stats.rs

```diff
@@ -0,0 +1,121 @@
+use anyhow::{Context, Result};
+use log::warn;
+use serde::{Deserialize, Serialize};
+use subprocess::{CaptureData, Exec, Result as PopenResult};
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Stats {
```

Comment:
Can we talk a bit more about sending these as options.  I understand why it useful for failure cases on the server-side, but I'm concerned about losing some of the type guarding we would otherwise get for free in the api.

Reply:
I don't think we're loosing any type guarding in API that way. `Options` says it's either `<T>` or `null`, therefore type check for `<T>` is still in place. By using `Option` I think we are getting both type check and error handling.